src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -135,13 +136,13 @@ void GcsServer::InitGcsActorManager() {
 }
 
 std::unique_ptr<rpc::JobInfoHandler> GcsServer::InitJobInfoHandler() {
-  return std::unique_ptr<rpc::DefaultJobInfoHandler>(
-      new rpc::DefaultJobInfoHandler(*redis_gcs_client_));
+  return std::unique_ptr<rpc::DefaultJobInfoHandler>(new rpc::DefaultJobInfoHandler(
```

Comment:
I think you can add a member of gcs::GcsPubSub to  `GcsServer` and pass it as a parameter to each handler.

Reply:
fixed